python/dgl/frame.py

```diff
@@ -53,108 +50,122 @@ def infer_scheme(tensor):
 class Column(object):
     """A column is a compact store of features of multiple nodes/edges.
 
-    Currently, we use one dense tensor to batch all the feature tensors
-    together (along the first dimension).
+    It batches all the feature tensors together along the first dimension
+    as one dense tensor.
+
+    The column can optionally have an index tensor I.
+    In this case, the i^th feature is stored in ``storage[index[i]]``.
+    The column class implements a Copy-On-Read semantics -- the index
+    select operation happens upon the first read of the feature data.
+    This is useful when one extracts a subset of the feature data
+    but wishes the actual index select happens on-demand.
 
     Parameters
     ----------
-    data : Tensor
-        The initial data of the column.
+    storage : Tensor
+        The feature data storage.
     scheme : Scheme, optional
         The scheme of the column. Will be inferred if not provided.
+    index : Tensor, optional
+        The row index to the feature data storage. None means an
+        identity mapping.
 
     Attributes
     ----------
+    storage : Tensor
+        Feature data storage.
     data : Tensor
-        The data of the column.
+        Feature data of this column.
```

Comment:
> In this case, I wonder whether distinguishing storage and data as two public attributes (and `index` as a public attribute) is necessary, especially since `storage` doesn't seem to be accessed anywhere outside `Column`.

I think your question is more about can we have only one public `data` attribute? For the moment, yes. But exposing them is also fine as long as we document them clearly. Will update the docstring.

Reply:
K.